File: app/src/main/res/ui/adpter/ItemTypeAdapter.java
package com.khedr_and_basuoin.deliveryapp.ui.adpter;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.khedr_and_basuoin.deliveryapp.databinding.ItemTypeBinding;
import com.khedr_and_basuoin.deliveryapp.databinding.ItemTypeshopBinding;
import com.khedr_and_basuoin.deliveryapp.pago.Meal;
import com.khedr_and_basuoin.deliveryapp.pago.ProductsType;


import org.jetbrains.annotations.NotNull;

import java.util.List;

public class ItemTypeAdapter extends RecyclerView.Adapter<ItemTypeAdapter.Popular> {
    Context context;
    List<ProductsType> list;
    Integer Item;
    static Integer Item2;


    public ItemTypeAdapter(Context context, Integer Item, List<ProductsType> list) {
        this.context = context;
        this.list = list;
        this.Item = Item;
        this.Item2 = this.Item;

    }


    @NonNull
    @NotNull
    @Override
    public Popular onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        return new ItemTypeAdapter.Popular(ItemTypeBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull Popular holder, int position) {

        holder.b.NameItmes.setText(list.get(position).getTitle());
        holder.b.CountItmes.setText(list.get(position).getCount()+" Items");
        holder.b.Product.setImageResource(list.get(position).getImages());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }


    public static final class Popular extends RecyclerView.ViewHolder {
        ItemTypeBinding b;
        public Popular(@NonNull ItemTypeBinding b) {
            super(b.getRoot());
            this.b = b;

        }
    }
}

File: app/src/main/java/com/khedrandbasuoni/baytifood/ui/adpter/ItemTypeAdapter.kt
package com.khedrandbasuoni.baytifood.ui.adpter

import android.annotation.SuppressLint
import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.khedrandbasuoni.baytifood.pojo.ProductsType
import com.khedrandbasuoni.baytifood.databinding.ItemTypeBinding

class ItemTypeAdapter(var context: Context, private var Item: Int, var list: List<ProductsType>) :
    RecyclerView.Adapter<ItemTypeAdapter.Popular>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Popular {
        return Popular(ItemTypeBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: Popular, position: Int) {
        holder.b.NameItmes.text = list[position].title
        holder.b.CountItmes.text = list[position].count.toString() + " Items"
        holder.b.Product.setImageResource(list[position].images)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class Popular(var b: ItemTypeBinding) : RecyclerView.ViewHolder(
        b.root
    )

    companion object {
        var Item2: Int = 0
    }

    init {
        Item2 = Item
    }
}